Sources/SDFCore/MacroInitializerRewriter.swift

```diff
@@ -142,20 +144,64 @@ private final class MacroInitializerRewriter: SyntaxRewriter {
         let infixOperand = InfixOperatorExprSyntax(foldedNode),
         infixOperand.leftOperand.trimmed.description.starts(with: "self.")
       {
-        assignments.append(
-          Assignment(infixOperand.leftOperand.description, statement))
+        let assignment = Assignment(
+          leftOperand: infixOperand.leftOperand.description,
+          item: statement,
+          group: .selfAssignments)
+        groupedAssignments.insert(assignment)
       } else {
-        otherStatements.append(statement)
+        let assignment = Assignment(
+          leftOperand: "",
```

Comment:
A normal statement doesn't have a concept of "leftOperand" and for this use case we don't need it (since we don't sort these guys).

I don't love the fact that we're doing these (I think it's a failure on the data structure I created), so I'm happy to hear your thoughts on how to improve this.

Reply:
Now that we are ensuring that the left and right side of the assignment are named the same thing, do we even need the left operand anymore? Could we just store and sort the entire statement?

Reply:
> Could we just store and sort the entire statement?

We could for the _sorting_ part, but we need the left operand information to know that this is a `self` assignment operation.

A high level [SequenceExprSyntax](https://swiftpackageindex.com/swiftlang/swift-syntax/601.0.0-prerelease-2024-11-04/documentation/swiftsyntax/sequenceexprsyntax) (which is what we get if we don't fold this statement) doesn't have a concept of "left" and "right" parts of an assignment (we need to make sure this is an infix operation first, so we know there's a left part of the operation and a right side), so I couldn't find a way to filter whether this is a `self.` assignment or not otherwise.

Happy to simplify this if you find a way though!

Reply:
Oh wait! You're right! I can just sort the items! Ok ignore my previous comment 😉

Reply:
https://github.com/fdiaz/SafeDI-Formatter/pull/8/commits/fcfd1d16483134cbcd1b626986514bf3f7d41f6a